pkg/controllers/osc/resources/cloud_provider.go

```diff
@@ -41,3 +47,14 @@ func GetCloudProviderFromMachineDeployment(md *clusterv1alpha1.MachineDeployment
 		Spec: *cloudProvider.CloudProviderSpec,
 	}, nil
 }
+
+func GetCloudConfig(secret *corev1.Secret) (string, error) {
+	if secret.Data == nil {
+		return "", fmt.Errorf("cloud-config secret data are empty")
+	}
+	cloudConfig := secret.Data[CloudConfigConfigName]
+	if len(cloudConfig) == 0 {
+		return "", fmt.Errorf("config not found in cloud-config secret data")
```

Comment:
You are not formatting any error so why using `fmt.Errorf` please use `errors.New()` instead